bench/tiered/push_pull.c

```diff
@@ -135,35 +143,110 @@ difftime_sec(struct timeval t0, struct timeval t1)
  *     This function runs the test for configured number of times to compute the average time taken.
  */
 static void
-run_test_clean(const char *suffix, uint64_t num_records, bool flush)
+run_test_clean(const char *suffix, uint32_t num_records, bool flush)
 {
     char home_full[HOME_BUF_SIZE];
-    double avg_time, avg_throughput;
-    int64_t avg_file_size;
+    double avg_wtime, avg_rtime, avg_wthroughput, avg_rthroughput;
+    uint64_t avg_file_size;
     int counter;
 
     avg_file_size = 0;
-    avg_time = avg_throughput = 0;
+    avg_wtime = avg_rtime = avg_rthroughput = avg_wthroughput = 0;
 
     for (counter = 0; counter < MAX_RUN; ++counter) {
         testutil_check(__wt_snprintf(
           home_full, HOME_BUF_SIZE, "%s_%s_%d_%d", opts->home, suffix, flush, counter));
         run_test(home_full, num_records, flush, counter);
+    }
 
-        /* Cleanup */
-        if (!opts->preserve)
-            testutil_clean_work_dir(home_full);
+    /* Cleanup */
+    if (!opts->preserve) {
+        testutil_check(__wt_snprintf(home_full, HOME_BUF_SIZE, "%s_%s*", opts->home, suffix));
+        testutil_clean_work_dir(home_full);
     }
 
+    /* Compute the average */
     for (counter = 0; counter < MAX_RUN; ++counter) {
-        avg_time += avg_time_array[counter];
-        avg_throughput += avg_throughput_array[counter];
+        avg_wtime += avg_wtime_arr[counter];
+        avg_rtime += avg_rtime_arr[counter];
+        avg_wthroughput += avg_wthroughput_arr[counter];
+        avg_rthroughput += avg_rthroughput_arr[counter];
         avg_file_size += avg_filesize_array[counter];
     }
 
-    printf("Bytes Written- %" PRIi64
-           " (~%s), Time took- %.3f seconds, Throughput- %.3f MB/second\n",
-      avg_file_size / MAX_RUN, suffix, avg_time / MAX_RUN, avg_throughput / MAX_RUN);
+    printf("Records in Bytes- %" PRIu64
+           " (~%s), W_Time- %.3f secs (SD- %.3f), W_Tput- %.3f MB/sec (SD- %.3f), R_Time- %.3f "
+           "secs (SD- %.3f), "
+           "R_Tput- %.3f MB/sec\n",
+      avg_file_size / MAX_RUN, suffix, avg_wtime / MAX_RUN, calculate_std_deviation(avg_wtime_arr),
+      avg_wthroughput / MAX_RUN, calculate_std_deviation(avg_wthroughput_arr), avg_rtime / MAX_RUN,
+      calculate_std_deviation(avg_rtime_arr), avg_rthroughput / MAX_RUN);
+}
+
+/*
+ * recover_verify --
+ *     Open wiredtiger and validate the data.
+ */
+static void
+recover_verify(const char *home, uint32_t num_records, uint64_t file_size, int counter)
```

Comment:
I wonder if you could change the name of this function?  Maybe `recover_validate`?  verify has a specific meaning in WT, and we're not calling verify.

Reply:
Updated the function name.